runtime/observability/logging-slf4j2/jvm/src/aws/smithy/kotlin/runtime/telemetry/logging/slf4j/Slf4j1xLoggerAdapter.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.telemetry.logging.slf4j
+
+import aws.smithy.kotlin.runtime.telemetry.context.Context
+import aws.smithy.kotlin.runtime.telemetry.logging.LogLevel
+import aws.smithy.kotlin.runtime.telemetry.logging.LogRecordBuilder
+import aws.smithy.kotlin.runtime.telemetry.logging.LoggerProvider
+import org.slf4j.MDC
+
+private val noOpLogRecordBuilder = LoggerProvider.None.getOrCreateLogger("NoOpLogger").atLevel(LogLevel.Error)
+
+/**
+ * SLF4J 1.x based logger
+ */
+internal class Slf4j1xLoggerAdapter(logger: org.slf4j.Logger) : AbstractSlf4jLoggerAdapter(logger) {
+    override fun atLevel(level: LogLevel): LogRecordBuilder = if (isEnabledFor(level)) {
+        Slf4j1xLogRecordBuilderAdapter(this, level)
+    } else {
+        noOpLogRecordBuilder
+    }
+}
+
+private class Slf4j1xLogRecordBuilderAdapter(
+    private val delegate: Slf4j1xLoggerAdapter,
+    private val level: LogLevel,
+) : LogRecordBuilder {
+
+    private var msg: (() -> String)? = null
+    private var cause: Throwable? = null
+    private val kvp by lazy { mutableMapOf<String, Any>() }
+    override fun setCause(ex: Throwable) {
+        cause = ex
+    }
+
+    override fun setMessage(message: String) {
+        msg = { message }
+    }
+
+    override fun setMessage(message: () -> String) {
+        msg = message
+    }
+
+    override fun setKeyValuePair(key: String, value: Any) {
+        kvp[key] = value
+    }
+
+    override fun setContext(context: Context) {
+        // TODO - add a way to get the current trace context and set the key/value pair on it?
+    }
```

Comment:
**Nit**: Could be moved down to `AbstractSlf4jLoggerAdapter`.

Reply:
Unfortunately it can't because this is on the `LogRecordBuilder` implementation which is the thing that differs between the two.

Reply:
Oh right, sorry I was thinking this was still in `Slf4j1xLoggerAdapter`. Nvm.